core-processor/src/ext.rs

```diff
@@ -477,16 +477,6 @@ impl BackendExternalities for Ext {
 }
 
 impl Ext {
-    fn check_message_value(&mut self, message_value: u128) -> Result<(), FallibleExtError> {
```

Comment:
Because of this check we had an invariant that any `program_actor -> user_actor` value transfer will be infallible. We still have with these changes this guarantee for `program_actor -> program_actor` interactions, as they always have ED on their balance and it's never touched. However, there are two cases worth testing:

1. program sends reply or just a message to user with a low gas (not enough for mailbox threshold) **with value(!)**.
2. program sends message **with value** to user and it reaches the mailbox.

There's an assumption that without this ED check transfers in `GearBank`/`Currency` from program to user can fail, i.e. cause runtime panic.

@breathx Can you double check it please?

Reply:
checked the first case - it can potentially panic btw

Reply:
Great catch

Transfer in journal handling in case of avoiding mailbox from program to user will cause panic if user doesnt exist and 0 < value < ed, so this case should be handled. Consider on error transfer in into unused bank funds

Reply:
Fixed